### DISCUSSION TITLE
Manga reader

### DISCUSSION THREAD

**Discussion by carif** [jellyfin/jellyfin-vue#1743] [Features]
Just posting some notes about manga's from matrix:

- Mangas can be saved in any ebook file format, like Comic Book Archive (cbz, cbr), EPUB, PDF etc.
- The ComicInfo metadata format for the Comic Book Archive format stores whether the comic is a manga or not. (the value is not yet read in by the bookshelf plugin, but it can be read in)
- Manga's are often read from right-to-left instead of left-to-right like western books and comics.
- Manga's **may** need some special handling for the EPUB format (see jellyfin/jellyfin-web#3345)
- Some manga have a panel that goes across multiple pages. In digital format, those pages may be stored in **landscape orientation** while the rest of the book is stored in portrait orientation. This breaks some ebook readers that are not designed to handle such a case.

---

The **Comic Book Archive** format:

This format is an archive that stores the pages as images. Usually, the archive file extension is changed, e.g. .zip -> .cbz or .rar -> .cbr. The format of the archive stays the same and can be opened by any application that can open a zip or rar file.
Common archive formats (from Wikipedia):

- .cb7 → [7z](https://en.wikipedia.org/wiki/7z)
- .cba → [ACE](https://en.wikipedia.org/wiki/ACE_(compression_file_format))
- .cbr → [RAR](https://en.wikipedia.org/wiki/RAR_(file_format))
- .cbt → [TAR](https://en.wikipedia.org/wiki/Tar_(file_format))
- .cbz → [ZIP](https://en.wikipedia.org/wiki/ZIP_(file_format))

The images that represent the pages can be stored in any file format, even .gif! They are usually numbered, e.g. `attackontitan_vol24_002.ext` for page 002 of that manga (`ext` being some arbitrary image file extension). Sometimes the cover page is named `cover.ext`.

Additional non-image files may be part of the Comic Book Archive to store additional information like metadata.

---

The **PDF** format:

Just classic PDF, nothing special here.

---

The **EPUB** format:

Most mangas I've seen use the format, that each page is represented with it's own `.xhtml` file that in turn links to an image file that is to be displayed. Each page also uses it's own style sheet or a shared one. Images are being loaded using the html-image tag inside of an html-svg tag (even when the loaded image is of type jpeg and not svg).
In my experience, the font is hard-coded within the images.

---

**The different formats a manga can be published in**:

1. [**Tankōbon**](https://en.wikipedia.org/wiki/Tank%C5%8Dbon) or collected volumes. Probably one of the most common digital ways to publish a manga. Those collect multiple chapters of a manga series and publish them in dedicated book-sized volumes (print and digital).
2. **Chapter based release**. Here, each chapter receives it's own release. This means, that instead of one file containing an entire book / volume, it contains only one chapter and multiple files together make up one book. While classic book stores might only sell manga's in Tankōbon form, there are dedicated websites that publish them chapter based where one can e.g. read the latest chapter for free (and has to pay for the other).
3. **Web manga**: Those are usually published one page at a time (e.g. one page per week).
4. **Magazines**: Magazines contain multiple manga series where each series has it's own slot of x pages to work with. If a lot of people like the series, then the series get's usually released in Tankōbon format as well.
5. **Webtoons** (as Wikipedia lists them under manga, I placed them here, but it's a digital comic that originated in South Korea):  Most of the time, Webtoons don't rely on the classic page approach, but instead let you scroll down each panel with panels being able to have different sizes within the same chapter. Some even include background audio played back either during the entire chapter or only played back on specific panels.

**Comment by Jstudner** [comment]
I would like to see this feature, I have webtoon formatted comics and they just don't work well as it is.
